xbmc/storage/DetectDVDType.cpp

```diff
@@ -108,84 +100,85 @@ void CDetectDVDMedia::UpdateDvdrom()
   // newly inserted media.
   {
     std::unique_lock<CCriticalSection> waitLock(m_muReadingMedia);
-    switch (GetTrayState())
+    switch (PollDriveState())
     {
-      case DRIVE_NONE:
+      case DriveState::DRIVE_NONE:
         //! @todo reduce / stop polling for drive updates
         break;
 
-      case DRIVE_OPEN:
+      case DriveState::DRIVE_OPEN:
+      {
+        // Send Message to GUI that disc been ejected
+        SetNewDVDShareUrl(CServiceBroker::GetMediaManager().TranslateDevicePath(m_diskPath), false,
+                          g_localizeStrings.Get(502));
+        CGUIMessage msg(GUI_MSG_NOTIFY_ALL, 0, 0, GUI_MSG_REMOVED_MEDIA);
+        CServiceBroker::GetGUI()->GetWindowManager().SendThreadMessage(msg);
+        // Clear all stored info
+        Clear();
+        // Update drive state
+        waitLock.unlock();
+        m_DriveState = DriveState::DRIVE_OPEN;
+        return;
```

Comment:
you should replace all the `return`s here with `break`s; Currently, there are both return and break for some cases which makes no sense.

Reply:
I know, but this part of the code was already like this (shows in the diff because of the indentation). As my next step will be focused on `DiscProbing` (which will likely make `DetectDVDType` class obsolete) I'd prefer to worry about this later. I feel it is out of scope of this PR original intent. I've been there already and know how this normally ends if we start changing a lot of stuff on the same PR :)